### PR TITLE
Add Boussinesq kernels for INSFV

### LINKED ISSUE #16755: INSFV Boussinesq
## Reason
We want to be able to model natural convection within the INSFV approximation

## Design
One new `FVElementalKernel`

## Impact
Ability to model natural convection with INSFV

### PR DESCRIPTION
**Pull request by lindsayad**
Closes #16755 

Comparison to results from #16023 look very good (and consistent with the article mentioned in #16023):

Ra = 1e6; streamlines (mesh uniformly refined twice)
ra1e6-streamlines

Ra = 1e5; streamlines
ra1e5-streamlines

Ra = 1e4; streamlines
ra1e4-streamlines

Ra = 1e3; streamlines
ra1e3-streamlines

Ra = 1e6; solid coloring velocity magnitude; temperature contours (mesh uniformly refined twice)
ra1e6-vel-mag-temp-contours

Ra = 1e5; solid coloring velocity magnitude; temperature contours
ra1e5-vel-mag-temp-contours

Ra = 1e4; solid coloring velocity magnitude; temperature contours
ra1e4-vel-mag-temp-contours

Ra = 1e3; solid coloring velocity magnitude; temperature contours
ra1e3-vel-mag-temp-contours

### CODE REVIEW

File: modules/navier_stokes/src/fv/kernels/INSFVGravityForce.C [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+registerMooseObject("NavierStokesApp", INSFVGravityForce);
+
+InputParameters
+INSFVGravityForce::validParams()
+{
+  InputParameters params = FVElementalKernel::validParams();
+  params.addClassDescription("Computes a body force due to gravity.");
+  params.addRequiredParam<RealVectorValue>("gravity", "Direction of the gravity vector");
+  MooseEnum momentum_component("x=0 y=1 z=2", "x");
```
Comment (GiudGiud) [inline modules/navier_stokes/src/fv/kernels/INSFVGravityForce.C:20]:
```suggestion
  MooseEnum momentum_component("x=0 y=1 z=2");
```
No need for a default if it's required?

File: modules/navier_stokes/doc/content/source/fv/kernels/INSFVBoussinesqBodyForce.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# INSFVBoussinesqBodyForce
+
+This object adds a $\rho\alpha\vec{g}(T - T_{ref})$ term to the incompressible
+Navier Stokes (INS) equations where $\rho$ is the density, $\alpha$ is the thermal
+expansion coefficient, $\vec{g}$ is the gravity vector, $T$ is the temperature,
+and $T_{ref}$ is a reference temperature. The term above introduces the
+Boussinesq approximation into the INS equations, which allows for modeling
+natural convection. This object must be added for every component
+of the velocity vector, e.g. $u$, $v$, $w$.
```
Comment (GiudGiud) [inline modules/navier_stokes/doc/content/source/fv/kernels/INSFVBoussinesqBodyForce.md:9]: I don't think you need it for u and v if g and w are aligned

File: modules/navier_stokes/doc/content/source/fv/kernels/INSFVGravityForce.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# INSFVGravityForce
+
+This object adds a simple gravitational force term to the incompressible Navier
+Stokes equations of the form $-\rho\vec{g}$ where $\rho$ is the density and
+$\vec{g}$ is the gravity vector. This object must be added for every component
+of the velocity vector, e.g. $u$, $v$, $w$.
```
Comment (GiudGiud) [inline modules/navier_stokes/doc/content/source/fv/kernels/INSFVGravityForce.md:6]: same

File: modules/navier_stokes/src/fv/kernels/INSFVBoussinesqBodyForce.C [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  InputParameters params = FVElementalKernel::validParams();
+  params.addClassDescription("Computes a body force for natural convection buoyancy.");
+  params.addRequiredCoupledVar("temperature", "temperature variable");
+  params.addRequiredParam<RealVectorValue>("gravity", "Direction of the gravity vector");
+  params.addParam<MaterialPropertyName>("alpha_name",
+                                        "alpha",
+                                        "The name of the thermal expansion coefficient"
+                                        "this is of the form rho = rho*(1-alpha (T-T_ref))");
+  params.addRequiredParam<Real>("ref_temperature", "The value for the reference temperature.");
+  MooseEnum momentum_component("x=0 y=1 z=2", "x");
```
Comment (GiudGiud) [inline modules/navier_stokes/src/fv/kernels/INSFVBoussinesqBodyForce.C:26]: same

File: modules/navier_stokes/doc/content/source/fv/kernels/INSFVBoussinesqBodyForce.md (hunk 2 above)
Comment (lindsayad) [reply modules/navier_stokes/doc/content/source/fv/kernels/INSFVBoussinesqBodyForce.md:9]: :+1:

File: modules/navier_stokes/src/fv/kernels/INSFVGravityForce.C (hunk 1 above)
Comment (lindsayad) [reply modules/navier_stokes/src/fv/kernels/INSFVGravityForce.C]: :+1:

**Comment by lindsayad** [comment]
Comments hopefully addressed

**Review by GiudGiud — APPROVED** [review]

**Comment by GiudGiud** [comment]
A little late, but I think we need kernels for the energy equation too
#worstreviewer

**Comment by GiudGiud** [comment]
Thoughts about renaming INSFVGravityBodyForce to INSFVMomentumGravity (no Body Force). It gets a little long with BodyForce?

**Comment by lindsayad** [comment]
What kernels do you think we need for the energy equation? I’m going to have to look back at the boussinesq benchmark equations but we are reproducing the benchmark results with the kernels we have...

**Comment by GiudGiud** [comment]
I'm thinking the work of these forces, so \vec{F} * \vec{v} terms. 
I would expect these terms to be small given the low speeds in the cavity

**Pull request merged**